### PR TITLE
Add sample service test rig and one simple test

### PR DESCRIPTION
**Pull request by MrCreosote**
Next PRs will fill out the tests. However, all the code changes that can be covered without bringing up dependent servers in  various states of usability are currently covered. Uncovered code is tested manually as it is not expected to change often.

Notes:

* focal caused the minio-related tests to fail. Not clear why this is.
* trying to install and use a python version that doesn't pre-exist in travis is a nightmare.
* trusty doesn't include python3.7 but xenial does. Bionic may work as well. Focal does not for the reason above.
* uwsgi 2.0.19 seemed to have build problems, so pinned to 2.0.18.
* Python's SSL module is broken on Trusty: deadsnakes/issues#63 . This prevented get-pip.py from working.

Everything that was tried to get Travis to finally work is preserved here: https://github.com/MrCreosote/workspace_deluxe/tree/python-travis-hell
This PR contains the final state of that branch.

### CODE REVIEW

**Review by briehl — COMMENTED** [review]
Just a couple minor documentation comments. The rest looks ok to me.

File: src/us/kbase/workspace/WorkspaceServer.java [hunk 1]
```diff
@@ -200,6 +202,16 @@ public DependencyStatus checkShockLink() {
 		}
 	}
 	
+	public DependencyStatus checkSampleLink() {
+		try {
```
[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/us/kbase/workspace/WorkspaceServer.java:206]: The happy path case is tested in the test rig below. The unhappy path test currently has to be tested manually... BUT in the next few PRs I refactor all the dependency checking code and it makes it possible to test the unhappy path with automated tests

File: src/us/kbase/workspace/kbase/InitWorkspaceServer.java [hunk 2]
```diff
@@ -398,23 +459,30 @@ private static AuthToken getKBaseToken(
 			final String source,
 			final ConfigurableAuthService auth)
 			throws WorkspaceInitException {
+		final AuthToken t = getToken(token, auth, source);
+		if (!t.getUserName().equals(user)) {
+			throw new WorkspaceInitException(String.format(
```
[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/us/kbase/workspace/kbase/InitWorkspaceServer.java:464]: Also the code rarely changes, so I live with testing manually when that occurs.

[the remark this answers is not in the document]
Comment (briehl) [reply src/us/kbase/workspace/kbase/InitWorkspaceServer.java:464]: Can you document somewhere how to do the manual testing?

[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/us/kbase/workspace/kbase/InitWorkspaceServer.java:464]: Hm, that's actually not trivial. how about an issue for now: https://github.com/kbase/workspace_deluxe/issues/371

[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/us/kbase/workspace/kbase/InitWorkspaceServer.java:464]: Also, that init code really needs a full blown rewrite. I'd prefer to do that first and then see what, if anything, still needs testing

**Comment by briehl** [comment]
Ok, all looks good. Thanks for the conversation.

**Pull request merged**